src/intel_pmu.c

```diff
@@ -283,6 +285,27 @@ static void pmu_submit_counter(int cpu, char *event, counter_t value) {
   plugin_dispatch_values(&vl);
 }
 
+meta_data_t *pmu_event_get_meta(struct event *e, int cpu) {
```

Comment:
You only ever seem to access `e->efd[cpu]` – can you pass that instead of `e` and `cpu` individually?